service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/AddressDetagQueue.kt

```diff
@@ -46,12 +46,25 @@ class AddressDetagQueue(
 
     override fun processMessage(message: AddressDetagDirective): AddressDetagOutcome {
         transaction {
-            AddressDeregistrationRecord.findForUpdate(message.id).first().let { addressDeregistration ->
+            AddressDeregistrationRecord.findPendingForUpdate(message.id).first().let { addressDeregistration ->
                 withMdc(*addressDeregistration.mdc()) {
-                    when (addressDeregistration.status) {
-                        INSERTED -> addressDetagService.createEvent(addressDeregistration)
-                        PENDING_TAG -> addressDetagService.eventComplete(addressDeregistration)
-                        else -> log.error("Invalid status - should never get here")
+                    val address = addressDeregistration.addressRegistration.address
+                    val existing = pbcService.getAttributeByTagName(address, bankClientProperties.kycTagName)
```

Comment:
wondering if these types of checks are really needed

if we batch broadcast a tx and we expect to read the result having committed through the event stream, shouldnt we assume its done when we see that tx commit? no need to do additional checks that something was done

Reply:
yeah I think you are right - checking the status of the tx hash should suffice